aten/src/ATen/templates/StorageDerived.cpp

```diff
@@ -37,8 +57,32 @@ static THAllocator storage_deleter = {
   nullptr,
   call_deleter,
 };
+static void* wrapped_alloc(void * ctx, ptrdiff_t size) {
```

Comment:
why is this a ptrdiff_t, but it's a size_t in the wrapped_alloc above?

Reply:
We're not consistent between THAllocator and THCDeviceAllocator. In `at::Allocator` it's `size_t`, which seems like the standard C++ type for this purpose.

Reply:
ya, this seems doubly strange because you can't allocate negative sizes and you would throw away that information anyway (because allocate takes size_t).  But I guess you are constrained here by TH/THC.